library/common-library/templates/_region.tpl

```diff
@@ -0,0 +1,74 @@
+{{/*
+Return the region that is being used by the user
+*/}}
+{{- define "newrelic.common.region" -}}
+{{- if and (include "newrelic.common.license._usesCustomSecret" .) (not (include "newrelic.common.region._fromValues" .)) -}}
+  {{- fail "This Helm Chart is not able to compute the region. You must specify a .global.region or .region if the license is set using a custom secret." -}}
+{{- end -}}
+
+{{- /* Defaults */ -}}
+{{- $region := "us" -}}
+{{- if include "newrelic.common.nrStaging" . -}}
+  {{- $region = "staging" -}}
+{{- else if include "newrelic.common.region._isEULicenseKey" . -}}
+  {{- $region = "eu" -}}
+{{- end -}}
+
+{{- include "newrelic.common.region.validate" (include "newrelic.common.region._fromValues" . | default $region ) -}}
+{{- end -}}
```

Comment:
Would it make sense to fail or add a warning when `include "newrelic.common.region._isEULicenseKey" .` returns `"true"` but `include "newrelic.common.region._fromValues" . | lower` is not `"eu"`?

Reply:
I wanted to keep the way the functions works consistent with how the `common-library` works: It computes things for you but then you can footgun yourself if you want to force something.

I also preferred this way to be consistent with the parsing of the license key. We cannot distinguish between a prod and a staging key, and we cannot fail if the key is not valid so I should not fail if the key and the region and the key does not match.